build.sh

```diff
@@ -2,43 +2,39 @@
 set -e
 
 # See https://stackoverflow.com/a/5148851
-require_clean_work_tree () {
-    git rev-parse --verify HEAD >/dev/null || exit 1
-    git update-index -q --ignore-submodules --refresh
-    err=0
+require_clean_work_tree() {
+  git rev-parse --verify HEAD >/dev/null || exit 1
+  git update-index -q --ignore-submodules --refresh
+  err=0
 
-    if ! git diff-files --quiet --ignore-submodules
-    then
-        echo >&2 "Cannot $1: You have unstaged changes."
-        err=1
-    fi
+  if ! git diff-files --quiet --ignore-submodules; then
+    echo >&2 "Cannot $1: You have unstaged changes."
+    err=1
+  fi
 
-    if ! git diff-index --cached --quiet --ignore-submodules HEAD --
-    then
-        if [ $err = 0 ]
-        then
-            echo >&2 "Cannot $1: Your index contains uncommitted changes."
-        else
-            echo >&2 "Additionally, your index contains uncommitted changes."
-        fi
-        err=1
+  if ! git diff-index --cached --quiet --ignore-submodules HEAD --; then
+    if [ $err = 0 ]; then
+      echo >&2 "Cannot $1: Your index contains uncommitted changes."
+    else
+      echo >&2 "Additionally, your index contains uncommitted changes."
     fi
+    err=1
+  fi
 
-    if [ $err = 1 ]
-    then
-        test -n "$2" && echo >&2 "$2"
-        exit 1
-    fi
+  if [ $err = 1 ]; then
+    test -n "$2" && echo >&2 "$2"
+    exit 1
+  fi
 }
 
 # Must be on master branch.
 BRANCH=$(git rev-parse --abbrev-ref HEAD)
-if [[ "$BRANCH" != "master" ]]; then
```

Comment:
I added this as a stopgap to have CI still be able to check that this can be built properly. This will be refactored later to have a separate script to actually do the release (which will enforce that we're only doing it from the main branch), so that we can remove the branch check from here